### PR TITLE
ci: Use the `--deployment` flag on `bundle install`

### PR DESCRIPTION
**Pull request by issyl0**
What
----

We can simplify our CI config further by [using the config from the
Actions conventions doc](https://github.com/alphagov/govuk-developer-docs/pull/2533).

How to review
-------------

Observe that CI still works!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,10 +1,7 @@
-name: Run tests
```
Comment (huwd) [inline .github/workflows/ci.yml:1]: How come we removed the name? Not needed?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,36 +18,27 @@ jobs:
     steps:
     - uses: actions/checkout@v1
     - uses: ruby/setup-ruby@v1
-
-    - name: Install PostgreSQL 11 client
```
Comment (huwd) [inline .github/workflows/ci.yml:25]: Don't we need postgres to run the tests?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -21,36 +18,27 @@ jobs:
...
       with:
         path: vendor/bundle
-        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
-        restore-keys: |
-          ${{ runner.os }}-gems-
+        key: bundle-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: bundle
 
-    - name: Build App
+    - run: sudo apt-get -yqq install libpq-dev
```
Comment (huwd) [inline .github/workflows/ci.yml:27]: Interesting, why move this above the env?

**Review by huwd — APPROVED** [review]
Left a couple of questions for bits i didn't understand but looks good

File: .github/workflows/ci.yml (hunk 3 above)
Comment (issyl0) [reply .github/workflows/ci.yml:27]: To keep the `uses` together.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (issyl0) [reply .github/workflows/ci.yml:1]: Yeah. We've broadly decided on no name in the conventions doc (see the link in the PR description).

**Pull request merged**